corehq/apps/es/case_search_bha.py

```diff
@@ -0,0 +1,41 @@
+from corehq.apps.es.case_search import CaseSearchES, ElasticCaseSearch
+
+from .cases import case_adapter
+from .client import create_document_adapter
+from .const import (
+    HQ_CASE_SEARCH_INDEX_CANONICAL_NAME,
+    HQ_CASE_SEARCH_BHA_INDEX_CANONICAL_NAME,
+    HQ_CASE_SEARCH_BHA_INDEX_NAME,
+    HQ_CASE_SEARCH_BHA_SECONDARY_INDEX_NAME,
+)
+from .index.settings import IndexSettingsKey
+
+
+class CaseSearchBhaES(CaseSearchES):
+    index = HQ_CASE_SEARCH_BHA_INDEX_CANONICAL_NAME
+
+
+class ElasticCaseSearchBha(ElasticCaseSearch):
+
+    settings_key = IndexSettingsKey.CASE_SEARCH_BHA
+    canonical_name = HQ_CASE_SEARCH_BHA_INDEX_CANONICAL_NAME
+    parent_index_name = HQ_CASE_SEARCH_INDEX_CANONICAL_NAME
+
+    def index(self, doc, refresh=False):
+        """
+        A ripoff of the `index` method in `ElasticDocumentAdapter`.
+        This is necessary because `ElasticCaseSearch` index is overridden
+        to index into both bha case search and the standard case search index.
+        """
+        doc_id, source = self.from_python(doc)
+        self._verify_doc_id(doc_id)
+        self._verify_doc_source(source)
+        self._index(doc_id, source, refresh)
```

Comment:
I really like the "sub indices" framing you added below.  If the parent can access the relationship in reverse, that might also be a nice way to fork migrations and writes, like the main `index` method could do something like this:
```python
        doc_id, source = self.from_python(doc)
        self._verify_doc_id(doc_id)
        self._verify_doc_source(source)
        self._index(doc_id, source, refresh)
        for sub_index in self.get_sub_indices():
            sub_index._index(doc_id, source, refresh)
```

Reply:
In my opinion, it would have made sense if we would have to always multiplex the writes. The writes are conditional based on domain and if the multiplex settings are configured. So it would look something like - 

```
        self._index(doc_id, source, refresh)
        for sub_index in self.get_sub_indices():
            if should_multiplex(source['domain']):
                sub_index._index(doc_id, source, refresh)
            
```

Thoughts?

Reply:
Yeah I like something like that - I was thinking the sub index should be the thing that decides whether to write or reject a doc, so that could be built in to its implementation of `_index`, or it could provide a `is_relevant` method or some such.

Reply:
That would make sense. Another thought, the current implementation can utilize `bulk` method to write in both the indices. 
The sub_indices approach would require us to make multiple queries which should be fine I think but definitely would be cleaner and more segregated.

Reply:
Ah that's a good point - I do think bulk writes make sense too.  That could be done by treating sub indices as contributors, something like:
```python
bulk_payload = self._get_payload(doc_id, source)
for index in sub indices:
    payload = index._get_payload(doc_id, source) 
    if payload:
        bulk_payload.extend(payload)
```